doc/flux_to_redux.md

```diff
@@ -0,0 +1,641 @@
+- __The State Class:__ we have a class (`ReduxState`) that has four properties: 
+    - mainBackgroundColor
+    - blockOneBackgroundColor
+    - blockTwoBackgroundColor
+    - blockThreeBackgroundColor
+- __Actions:__ we have four actions as their own classes; one class for each state class property.
+- __The Reducer:__ as probably expected, our reducer is also similar to the simple app but with a condition for each action. This reducer is especially contrived because the simplicity of the store data makes each case the same with the only difference being the property the update is pointed at. A more complex usage could leverage [Combine Reducers](https://github.com/johnpryan/redux.dart/blob/master/doc/combine_reducers.md).
+- __The Store Object:__ same as the simple example.
+- __The UI:__ like the simple example, our component tree is wrapped in a `ReduxProvider`. Otherwise the component architecture is the same as the Flux version, minus the Flux-y parts. We have a connected `UiComponent2` called `BigBlock` that maps the store state to props, which passes the props to its three `LittleBlock`s, with four buttons to trigger the background actions. Naturally all the Flux parts were removed, including the `redrawOn` override.
+
+```dart
+// A regular Dart class that represents the application data model
+class ReduxState {
+    // All of the state fields for the application
+    String mainBackgroundColor;
+    String blockOneBackgroundColor;
+    String blockTwoBackgroundColor;
+    String blockThreeBackgroundColor;
+
+    // A constructor for creating the default app state
+    ReduxState.defaultState()
+        : this.mainBackgroundColor = 'gray',
+        this.blockOneBackgroundColor = 'red',
+        this.blockTwoBackgroundColor = 'orange',
+        this.blockThreeBackgroundColor = 'blue';
+
+    // A convenience constructor for creating new instances of the state class
+    ReduxState.update(ReduxState oldState,
+        {mainBackgroundColor, blockOneBackgroundColor, blockTwoBackgroundColor,         blockThreeBackgroundColor})
+        : this.mainBackgroundColor = mainBackgroundColor ?? oldState.mainBackgroundColor,
+        this.blockOneBackgroundColor = blockOneBackgroundColor ?? oldState.blockOneBackgroundColor,
+        this.blockTwoBackgroundColor = blockTwoBackgroundColor ?? oldState.blockTwoBackgroundColor,
+        this.blockThreeBackgroundColor = blockThreeBackgroundColor ?? oldState.blockThreeBackgroundColor;
+}
+```
+
+```dart
+// An action for updating each state property
+class UpdateBackgroundColorAction {}
+
+class UpdateBlockOneBackgroundColorAction {}
+
+class UpdateBlockTwoBackgroundColorAction {}
+
+class UpdateBlockThreeBackgroundColorAction {}
+```
+
+```dart
+// A simple reducer that has a case for every action.
+// Because the example is relatively contrived to keep things simple, there are other
+// ways this could be done given that all the actions are pretty much the same,
```

Comment:
if it's easy, you could include both this as the verbose option and an equivalent shortened version

Reply:
actually if you're just talking about bubbling the `var color` line outside of the if statements or breaking it out into a separate utility method (my preference), I don't think it's any harder to grok with that change so you could probably stick with the short version

Reply:
I was actually thinking that the actions and reducer together could be refactored to really only require one action - just an `UpdateColorAction` that takes in all the colors, including one that's changed, and passes them all in to the `update` constructor. That was really unclear though, so I updated the wording to be a little more specific as to what 'simplified' meant.

I had thought about breaking color generation into a different method though, and was just on the fence originally. So I updated that as well!